package.json

```diff
@@ -92,6 +92,12 @@
                     "scope": "resource",
                     "type": "array"
                 },
+                "black-formatter.show-formatting-messages": {
```

Comment:
@luabud Can you look at the naming of this setting and description for this? This setting allows users to get a visible UI notification if there is a issue when formatting. We also need to decide on the default value for this setting.

I feel like this should be a enum, with values like `never`, `on-error`, and `on-warn`. `notify-on-error` covers scenarios where formatting fails due to some error like syntax error, or out-of-date version etc. `notify-on-warn` covers scenarios in the `notify-on-error` plus cases where formatting is intentionally skipped.

Reply:
May be `black-formatter.showNotification`? so that `"black-formatter.showNotification": "on-warn"` makes more sense as to what it will do.

@luabud what do you think the default should be?

Reply:
Love `showNotification`! As for the default value, my vote would be for "on-error" to avoid annoying those who feel strongly about notifications. But I don't have a strong opinion on this 😊

Reply:
@luabud, Initially I agreed that default `"on-error"` would make sense, but after some thoughts I'd vote for `"on-warn"`. The reason is simple: if user is annoyed by notifications, it's clear what's happening and how to fix it. On contrary, when extension doesn't do what it supposed to do and doesn't emit any warnings either, it's unclear what's happening and how to fix it.